File: service/src/main/java/com/loudsight/utilities/service/dispatcher/MessageHandler.java
package com.loudsight.utilities.service.dispatcher;

import com.loudsight.useful.entity.permission.Subject;
import com.loudsight.utilities.service.dispatcher.bridge.BridgeMessageType;

public interface MessageHandler<A> {
    A onMessage(Address to,
                       Address replyTo,
                       Subject recipient,
                       Subject sender,
                       BridgeMessageType publicationType,
                       Object payload);
}

File: service/src/main/java/com/loudsight/utilities/service/dispatcher/MessageHandler.kt
package com.loudsight.utilities.service.dispatcher

import com.loudsight.utilities.permission.Subject
import com.loudsight.utilities.service.dispatcher.bridge.BridgeMessageType

interface MessageHandler<A> {
    fun onMessage(to: Address,
                  replyTo: Address,
                  recipient: Subject,
                  sender: Subject,
                  publicationType: BridgeMessageType,
                  payload: Any?): A
}